src/reducers/PatientReducer.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import { PATIENT_ACTIONS } from '../actions/PatientActions';
+
+const patientInitialState = () => ({
+  currentPatient: null,
```

Comment:
I wondered about having a 'validity' property like 
```
const patientInitialState = () => ({
  currentPatient: null,
  firstName: '',
  lastName: '',
  code: '',
  dateOfBirth: '',
  email: '',
  phone: '',
  addressOne: '',
  addressTwo: '',
  country: '',
  isValid: {
    firstName: true,
    lastName: true,
    code: true,
    dateOfBirth: true,
    email: true,
    phone: true,
    country: true,
  },
});
```

which might make the `fieldLookup` below redundant?
`      return { ...state, isValid: { [field]: validity } }; `

( alternative thought was to have the shape `{ value: '', isValid: true }` for each property which I also like )

Just feels like a lot of duplication having properties doubled up with their 'IsValid' version

Reply:
much better :) (just have the values which have some sort of constraint currently in desktop.. although that's a poor excuse to not have better validation on some of these fields)
fa9b01e

Reply:
Oh, nice, I like what you've done with pulling out `isValid` and adding in `newValidityState`